### PR TITLE
Pipeline: remove command propagation for IPC4

### PR DESCRIPTION
**Pull request by lyakh**
This might fix at least some cases of #6220

### CODE REVIEW

File: src/audio/pipeline/pipeline-stream.c [hunk 1]
```diff
@@ -154,7 +155,7 @@ static int pipeline_comp_list(struct comp_dev *current,
 	bool is_same_sched = pipeline_is_same_sched_comp(current->pipeline,
 							 ppl_data->start->pipeline);
 
-	if (!is_single_ppl && !is_same_sched) {
+	if (!is_single_ppl && (!is_same_sched || IPC4_MOD_ID(current->ipc_config.id))) {
```
Comment (lgirdwood) [inline src/audio/pipeline/pipeline-stream.c:158]: getting complex, best to have a comment to explain.

File: src/include/ipc4/module.h [hunk 2]
```diff
@@ -24,6 +24,8 @@
 #ifndef __IPC4_MODULE_H__
 #define __IPC4_MODULE_H__
 
+#include <ipc4/error_status.h>
+
```
Comment (kv2019i) [inline src/include/ipc4/module.h:28]: Unrelated?

File: src/audio/pipeline/pipeline-graph.c [hunk 3]
```diff
@@ -348,6 +349,9 @@ static int pipeline_comp_reset(struct comp_dev *current,
 	/* reset should propagate to the connected pipelines,
 	 * which need to be scheduled together
 	 */
+	if (!is_single_ppl && IPC4_MOD_ID(current->ipc_config.id))
+		return 0;
```
Comment (kv2019i) [inline src/audio/pipeline/pipeline-graph.c:352]: Maybe explain in the commit what's the logic behind the "IPC4_MOD_ID()" check. Without context, this seems hard to understand.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Fix is good but a few readability comments.

File: src/include/ipc4/module.h (hunk 2 above)
Comment (lyakh) [reply src/include/ipc4/module.h:28]: It is related. This header is now included in pipeline files from this patch and that breaks the build because this header uses macros, defined in error_status.h. This works before this patch because that header gets included via other paths currently, but since this header needs it, it should be included directly.

File: src/audio/pipeline/pipeline-graph.c (hunk 3 above)
Comment (lyakh) [reply src/audio/pipeline/pipeline-graph.c:354]: Updated

File: src/audio/pipeline/pipeline-stream.c (hunk 1 above)
Comment (lyakh) [reply src/audio/pipeline/pipeline-stream.c:163]: Done

**Review by kv2019i — APPROVED** [review]

**Comment by lgirdwood** [comment]
@marcinszkudlinski @mwasko potential fix for #6351 (needs to be checked).
